Documentation for release (#337)
codeplaysoftware/cutlass-sycl: media/docs/build/building_with_sycl_support.md

To build CUTLASS SYCL support for Intel GPUs, you need the DPC++ compiler;
you can use the latest [[nightly build](https://github.com/intel/llvm/releases)] 

> or a oneAPI toolkit from 2025.1 onwards. The Intel Graphics Compiler (minimum version ????) is also required.

Reviewer: ```suggestion
or a oneAPI toolkit from 2025.1 onwards. The next public release of the  Intel Graphics Compiler is also required.
```

  -DCMAKE_CXX_FLAGS="-ftemplate-backtrace-limit=0 -fdiagnostics-color=always"  
```

> Note: When building with SYCL support, CMake currently sets the following environment variable to work around an IGC bug:

Reviewer 2: On the `Requirements (SYCL for Intel GPU)` section in this file, update the OneAPI version to 2025.1, also add the requirement for the IGC version (@mehdi-goli to provide the IGC version)

and it will download the MKL library if it cannot find it.

> 

Reviewer: ```suggestion
To get better performance result we require the following combinations of the environment variables flags to provide better performance hints for generating optimised code.
```bash
 export SYCL_PROGRAM_COMPILE_OPTIONS="-ze-opt-large-register-file -gline-tables-only" 
export ONEAPI_DEVICE_SELECTOR=level_zero:gpu
export IGC_VISAOptions="-perfmodel"
export IGC_VectorAliasBBThreshold=10000
export IGC_ExtraOCLOptions="-cl-intel-256-GRF-per-thread"  
```

Reviewer 3: - `-gline-tables-only` isn't for performance
- we should mention that for AOT those options have to be set during compilation and for JIT when running

To build and run a simple PVC gemm example run the commands below.

```
$ ninja 00_pvc_gemm

> $ ./examples/sycl/00_pvc_gemm/00_pvc_gemm

Reviewer: ```suggestion
$ ONEAPI_DEVICE_SELECTOR=level_zero:gpu ./examples/sycl/00_pvc_gemm/00_pvc_gemm
```

Reviewer 3: why do you suggest that. level_zero:gpu is the first pick anyhow so this shouldn't change anything and makes it more verbose for the user.

Reviewer: Yes that is correct, I wanted to emphasise on the GPU selection if the first pick for some reason it is not GPU
